src/language/language.rs

```diff
@@ -1487,7 +1451,10 @@ impl egg::Analysis<Language> for MyAnalysis {
                             .map(|id| &egraph[*id].data)
                             .collect::<Vec<_>>()[..]
                         {
-                            [MyAnalysisData::AccessPattern(a), MyAnalysisData::AccessPattern(_), MyAnalysisData::Legacy(_)] => {
+                            [MyAnalysisData::AccessPattern(a), MyAnalysisData::AccessPattern(_), MyAnalysisData::Usize(_)] => {
+                                a.clone()
+                            }
+                            [MyAnalysisData::AccessPattern(a), MyAnalysisData::AccessPattern(_), MyAnalysisData::Shape(_)] => {
```

Comment:
```suggestion
                            [MyAnalysisData::AccessPattern(a), MyAnalysisData::AccessPattern(_), MyAnalysisData::Usize(_) | MyAnalysisData::Shape(_)] => {
```

Reply:
Can you try this, here and elsewhere? You can use the `|` operator to match multiple things. I think this is valid here, but I'm not sure.

Reply:
There are other places in this PR where you can make this same change -- can you track them down?

Reply:
eg line 1695

Reply:
I think I should have gotten all of them now, I found one more place that I could make the change as of writing this comment,  so it should be good now?